present-day-forecasting/main.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import sys
+from pathlib import Path
+
+import jax
+import numpy as np
+import numpyro
+import polars as pl
+import yaml
+from numpyro.infer import MCMC, NUTS
+
+import linmod.eval
+import linmod.models
+from linmod.visualize import plot_forecast
+
+numpyro.set_host_device_count(4)
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs, file=sys.stderr)
+    sys.stderr.flush()
+
+
+# Load configuration
+
+if len(sys.argv) != 2:
+    log("Usage: python3 main.py <YAML config path>")
+    sys.exit(1)
+
+with open(sys.argv[1]) as f:
+    config = yaml.safe_load(f)
+
+# Load the data
+
+data = pl.read_csv(config["data"]["save_path"], try_parse_dates=True)
+
+# Fit each model
+
+forecast_dir = Path(config["forecasting"]["save_path"])
+forecast_dir.mkdir(exist_ok=True)
+
+for model_name in config["forecasting"]["models"]:
+    forecast_path = forecast_dir / f"forecasts-{model_name}.csv"
```

Comment:
Aside from python modules and dates, CFA style is to use underscores in file names (snake case): https://github.com/cdcent/cfa-predict-handbook/blob/main/docs/style_guide.md#file-names

Reply:
So the ideal here would be `forecasts_baseline_model.csv` but I can see arguments for having the model name be considered a single character, in which case I suppose some violating of the norms makes sense.

But I think `forecasts_ModelName.csv` is still technically preferable to `forecasts-ModelName.csv`